src/Input/Select/Select.tsx

```diff
@@ -15,346 +15,287 @@ import {
         onOptionClick(e);
       }
-    } else if (e.keyCode === 27) {
-      inputElement.blur();
-      this.setState({ isFocus: false });
-    }
-  };
+    },
+    [children, onChange, filterValue, getActiveElement]
+  );
 
-  handleMouseEnter = (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
-    const listItemElement = e.target as HTMLLIElement;
-    this.setState({
-      cursor: Number(listItemElement.dataset.id),
-    });
-  };
-
-  scrollToActiveElement = () => {
-    const selectListElement = this.node.current.querySelector(
+  const scrollToActiveElement = React.useCallback(() => {
+    const selectListElement = node.current.querySelector(
       '.select-listbox'
     ) as HTMLUListElement;
-    selectListElement.scrollTop = this.getActiveElement().offsetTop;
-  };
-
-  getActiveElement = () => {
-    return this.node.current.querySelector('.active') as HTMLLIElement;
-  };
-
-  render() {
-    const {
-      label,
-      status,
-      disabled,
-      className,
-      onFocus,
-      onBlur,
-      onChange,
-      noOptionResult = 'No results found',
-      small,
-      disableTyping,
-      removeFloatingLabel,
-      removeDropIcon,
-      error,
-      renderError,
-      value, // eslint-disable-line @typescript-eslint/no-unused-vars
-      children, // eslint-disable-line @typescript-eslint/no-unused-vars
-      ...defaultProps
-    } = this.props;
-
-    const {
-      floating,
-      isFocus,
-      selectedValue = '',
-      filterValue,
-      cursor,
-      isLoading,
-    } = this.state;
-
-    const deprecatedStatus = status || (error && 'error');
-
-    const shouldShowError = error && typeof error !== 'boolean';
-
-    let completeError = null;
-    if (shouldShowError) {
-      if (renderError) {
-        completeError = renderError(error);
-      } else {
-        completeError = (
-          <SelectErrorDefault className="select-error-default">
-            {error}
-          </SelectErrorDefault>
-        );
+    selectListElement.scrollTop = getActiveElement().offsetTop;
+  }, [getActiveElement]);
+
+  // Should be called when any key is pressed inside the component
+  const handleKeyDown = React.useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      const inputElement = e.target as HTMLInputElement;
+
+      if (disableTyping) {
+        e.preventDefault();
       }
+
+      if (e.keyCode === 38 && cursor > 0) {
+        setCursor(cursor - 1);
+        setShouldScrollToCursor(true);
+      } else if (e.keyCode === 40 && cursor < filterValue.length - 1) {
+        setCursor(cursor + 1);
+        setShouldScrollToCursor(true);
+      } else if (e.keyCode === 13) {
+        inputElement.blur();
+        handleSelect();
+      } else if (e.keyCode === 27) {
+        inputElement.blur();
+        setIsFocus(false);
+      }
+    },
+    [disableTyping, filterValue, cursor, handleSelect]
+  );
+
+  React.useEffect(() => {
+    if (cursor && shouldScrollToCursor) {
```

Comment:
I think there must be a condition is `cursor` is zero and `shouldScrollToCursor ` is true. You can test with long options, you won't be able to scroll to first options.

Reply:
Removed!